tests/python_tests/test_continuous_batching.py

```diff
@@ -374,7 +373,27 @@ def test_pipelines_generate_with_streaming(tmp_path, pipeline_type):
     models_path : Path = tmp_path / "t_streaming" / model_id
     convert_models(opt_model, hf_tokenizer, models_path)
 
-    pipe, input, gen_config = get_data_by_pipeline_type(models_path, pipeline_type)
+    generation_config = GenerationConfig()
+    pipe, input, gen_config = get_data_by_pipeline_type(models_path, pipeline_type, generation_config)
+    py_streamer = lambda x: False
+    _ = pipe.generate(input, generation_config=gen_config, streamer=py_streamer)
+
+    del pipe
+    rmtree(models_path)
+
+@pytest.mark.parametrize("pipeline_type", ["continuous_batching", "speculative_decoding", "prompt_lookup_decoding", "llm_pipeline"])
+@pytest.mark.precommit
+def test_pipelines_generate_with_streaming_empty_output(tmp_path, pipeline_type):
+    model_id : str = "facebook/opt-125m"
+    opt_model, hf_tokenizer = get_hugging_face_models(model_id)
+
+    models_path : Path = tmp_path / "t_streaming" / model_id
+    convert_models(opt_model, hf_tokenizer, models_path)
+    
+    generation_config = GenerationConfig()
+    generation_config.stop_strings = {" the "}
+
+    pipe, input, gen_config = get_data_by_pipeline_type(models_path, pipeline_type, generation_config)
```

Comment:
Can we confirm that we indeed got empty output here (or just a stop string if `include stop string` is set) to make sure this test is checking correct scenario?

Reply:
Honestly, we can not check your case (with step + read using) via generate, because in `generate` method we are using `has_active_requests` simultaneously with flag. Let me think about the another way to check it

Reply:
we can check that callback is never called => it means stop_str matches with generated output

Reply:
Can't we just check output and compare with expected? 
If we run generate with stop string `" the "`  then if we include stop string in output, we can check if generated text has only stop string, or if we don't include, then check if generated text is empty. Would that work?

Reply:
Of  course we can, but this is not a test to check the streaming. Moreover, in client scenario (generate) we won't have hung according threading logic and conditional variable using (we will check)
```
cv.wait(lock, [&generation, &has_active_requests]{
    return generation->can_read() || !has_active_requests;
});`)
```
So, we can not cover using of `step() + read()` like in ovms... 
But we with @ilya-lavrenov  generated a workaround:
Streamer callback will be called in case of valid token generation. So, If we will generate empty output, we won't call callback. => throwing an exception from streamer will roune test at all in case of calling. But the expected result is `` => callback won't be called at all

Reply:
Okay. This behavior is also tested in OVMS validation so there is coverage for step() + read() scenario.